Dockerfile

```
@@ -1,10 +1,16 @@
-FROM python:3.6-slim-stretch
+# syntax=docker/dockerfile:1.4
 
+FROM scratch AS installer
 COPY ./ /var/cache/napalm/
 
-RUN apt-get update \
- && apt-get install -y python-dev python-cffi libxslt1-dev libssl-dev libffi-dev \
- && apt-get autoremove \
- && rm -rf /var/lib/apt/lists/* \
- && pip --no-cache-dir install -U cffi cryptography /var/cache/napalm/ \
- && rm -rf /var/cache/napalm/
+FROM python:3.9-slim
```

```suggestion
FROM python:3.12-slim-bookworm
```

The reason why it's a good idea to specify the distro (as above) is because `3.9-slim` currently defaults to Bookworm and therefore the build fails because `python3.9-dev isn't available`:

```
#11 2.845 Building dependency tree...
#11 2.954 Reading state information...
#11 3.051 E: Unable to locate package python3.9-dev
#11 3.051 E: Couldn't find any package by glob 'python3.9-dev'
#11 3.051 E: Couldn't find any package by regex 'python3.9-dev'
------
```

We can probably bump to Python 3.12 which NAPALM is now compatible with.